java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -117,8 +117,7 @@ protected void reallocOffsetBuffer() {
 
     final ArrowBuf newBuf = allocator.buffer((int) newAllocationSize);
     newBuf.setBytes(0, offsetBuffer, 0, currentBufferCapacity);
-    final int halfNewCapacity = newBuf.capacity() / 2;
-    newBuf.setZero(halfNewCapacity, halfNewCapacity);
+    newBuf.setZero(currentBufferCapacity, newBuf.capacity() - currentBufferCapacity);
```

Comment:
`newAllocationSize` can be more than `currentBufferCapacity * 2`. But we just set zeros since `halfNewCapacity`, so I think previously we miss to zero out since `halfNewCapacity` to `newAllocationSize / 2`. Is it?